core/interfaces.go

```diff
@@ -27,101 +29,101 @@ import (
 // * One path for certificates -> Cert
```

Comment:
These now have a mix of methods with named and unnamed arguments, could you name them all.